src/test/regression/CMakeLists.txt

```diff
@@ -34,3 +34,11 @@ add_shadow_tests(
       # the full timeout to fail otherwise.
       TIMEOUT 5
     )
+
+# Make sure tests in subdirs append to previously collected tests.
+set(ALL_SHADOW_TESTS "${ALL_SHADOW_TESTS}")
```

Comment:
Maybe I'm misremembering how cmake scoping works, but I think this is a no-op?

Reply:
I have no idea, this is how it was done in #1131. @robgjansen Do you remember why you did it this way 2 years ago? :)

Reply:
Without doing this the subdir tests would drop out of scope in the parent directory, meaning that the parent's list of tests would not include the subdir tests and they wouldn't run. Maybe that changed with a newer cmake version, but it would be pretty easy to test what happens if you don't include this blurb.

Reply:
I removed it and the list of tests printed during the `setup build --test` don't change, so I'm guessing it's not doing anything on the version of cmake that I'm using.

I'll add another commit to remove this line everywhere we do it.

Reply:
It looks like the only place we use this variable is to print a count of the number of tests?

```
$ git grep ALL_SHADOW_TESTS
src/test/CMakeLists.txt:   set(ALL_SHADOW_TESTS "${ALL_SHADOW_TESTS}" PARENT_SCOPE)
src/test/CMakeLists.txt:   list(APPEND ALL_SHADOW_TESTS ${SHADOW_TEST_NAME})
src/test/CMakeLists.txt:list(LENGTH ALL_SHADOW_TESTS ALL_SHADOW_TESTS_LENGTH)
src/test/CMakeLists.txt:message(STATUS "Configured to build ${ALL_SHADOW_TESTS_LENGTH} Shadow tests.")
src/test/CMakeLists.txt:message(STATUS "ALL_SHADOW_TESTS = ${ALL_SHADOW_TESTS}")
src/test/config/CMakeLists.txt:set(ALL_SHADOW_TESTS "${ALL_SHADOW_TESTS}")
src/test/config/CMakeLists.txt:set(ALL_SHADOW_TESTS "${ALL_SHADOW_TESTS}" PARENT_SCOPE)
src/test/examples/CMakeLists.txt:set(ALL_SHADOW_TESTS "${ALL_SHADOW_TESTS}")
src/test/examples/CMakeLists.txt:set(ALL_SHADOW_TESTS "${ALL_SHADOW_TESTS}" PARENT_SCOPE)
src/test/socket/CMakeLists.txt:set(ALL_SHADOW_TESTS "${ALL_SHADOW_TESTS}")
src/test/socket/CMakeLists.txt:set(ALL_SHADOW_TESTS "${ALL_SHADOW_TESTS}" PARENT_SCOPE)
src/test/tgen/CMakeLists.txt:set(ALL_SHADOW_TESTS "${ALL_SHADOW_TESTS}")
src/test/tgen/CMakeLists.txt:set(ALL_SHADOW_TESTS "${ALL_SHADOW_TESTS}" PARENT_SCOPE)
src/test/tor/CMakeLists.txt:set(ALL_SHADOW_TESTS "${ALL_SHADOW_TESTS}")
src/test/tor/CMakeLists.txt:set(ALL_SHADOW_TESTS "${ALL_SHADOW_TESTS}" PARENT_SCOPE)
```

Reply:
Hmm, okay I'll leave things as-is then, and make a new PR to remove this list.